### PR TITLE
Recipe Adder v2

### PR DESCRIPTION
**Pull request by Glease**
WIP

- [x] Builder for recipe
- [x] Add doAdd() for all GT5 recipe maps
- [ ] Add recipe validation 
	Partially done. Need to figure out how many input a recipe map can accept.
- [ ] Remove unused feature and code generated during prototyping.
- [x] Migrate a few recipe script files to RA2
- [x] ~~Add a way to automatically migrate old recipe script file to RA2 due to how great everything changed.
	Hint: `java.lang.reflect.Proxy` might be helpful. In general just replace GT_Values.RA with a custom built one, that, while adding recipe, also tries to figure out which class and method this recipe originates from, and what arguments was passed, then dump them into a a) machine readable form b) .java directly c) java snippets that can be semi automatically adapted to actual recipe script.~~ done by boubou

### CODE REVIEW

**Comment by chochem** [comment]
@Glease can you rebase to include https://github.com/GTNewHorizons/GT5-Unofficial/commit/ea95663045633c7d0caa584de3aefc9f01176d29 ?
assembler and assembling line are not touched here yet so I think it should be fine

File: src/main/java/gregtech/loaders/postload/recipes/VacuumFreezerRecipes.java [hunk 1]
```diff
@@ -18,241 +17,672 @@ public class VacuumFreezerRecipes implements Runnable {
...
+
+                GT_Values.RA.stdBuilder()
+                    .itemInputs(
+                        GT_ModHandler.getIC2Item("reactorHeatSwitchSpread", 1L, 32767)
+                    )
+                    .itemOutputs(
+                        GT_ModHandler.getIC2Item("reactorHeatSwitchSpread", 1L, 1)
+                    )
+                    .noFluidInputs()
+                    .noFluidOutputs()
```
[the remark this answers is not in the document]
Comment (boubou19) [reply src/main/java/gregtech/loaders/postload/recipes/VacuumFreezerRecipes.java:57]: it's a choice of mine, so recipes look more uniform. At first glance you can understand what's going on, no need to have preexisting knowledge on internal defaults. It also doesn't cost me anything to add it, as most of the recipes i commit are auto generated from the previous ones.

**Comment by boubou19** [comment]
progress:

- [ ] AlloySmelter
- [ ] ArcFurnace
- [ ] Assembler
- [ ] AssemblyLine
- [ ] Autoclave
- [ ] Bender
- [ ] BlastFurnace
- [ ] Brewery
- [ ] Canner
- [ ] Centrifuge
- [ ] ChemicalBath
- [ ] Chemical
- [ ] CircuitAssembler
- [ ] Compressor
- [ ] CropProcessing
- [ ] Cutting
- [ ] Distillery
- [ ] Electrolyzer
- [ ] ElectromagneticSeparator
- [ ] Extractor
- [ ] Extruder
- [ ] Fermenter
- [ ] FluidCanner
- [ ] FluidExtractor
- [ ] FluidHeater
- [ ] FluidSolidifier
- [ ] ForgeHammer
- [ ] FormingPress
- [ ] Fuel
- [ ] FusionReactor
- [ ] ImplosionCompressor
- [ ] LaserEngraver
- [ ] Lathe
- [ ] MatterAmplifier
- [ ] Mixer
- [ ] NEIHiding
- [ ] OreDictUnification
- [ ] Packager
- [ ] PlasmaForge
- [ ] Printer
- [ ] Pulverizer
- [ ] Pyrolyse
- [ ] RecipeRemover
- [ ] Sifter
- [ ] Slicer
- [ ] Smelter
- [ ] Thaumcraft
- [ ] ThermalCentrifuge
- [ ] VacuumFreezer
- [ ] Wiremill
- [ ] TranscendentPlasmaMixer

**Comment by eigenraven** [comment]
From github actions log:
```
Time: 4/1/23 2:18 PM
Description: Exception in server tick loop

java.lang.NullPointerException: Exception in server tick loop
	at gregtech.api.util.GT_Recipe$GT_Recipe_Map.lambda$static$0(GT_Recipe.java:1667)
	at gregtech.api.util.GT_Recipe$GT_Recipe_Map.lambda$withIdentityReturn$13(GT_Recipe.java:3398)
	at gregtech.api.util.GT_Recipe$GT_Recipe_Map.doAdd(GT_Recipe.java:3653)
	at gregtech.api.util.GT_RecipeBuilder.addTo(GT_RecipeBuilder.java:647)
	at gregtech.loaders.postload.recipes.DistilleryRecipes.addUniversalDistillationRecipe(DistilleryRecipes.java:990)
	at gregtech.loaders.postload.recipes.DistilleryRecipes.universalDistillationTowerRecipes(DistilleryRecipes.java:653)
	at gregtech.loaders.postload.recipes.DistilleryRecipes.run(DistilleryRecipes.java:27)
	at gregtech.loaders.postload.GT_MachineRecipeLoader.run(GT_MachineRecipeLoader.java:41)
	at gregtech.GT_Mod.onPostLoad(GT_Mod.java:379)
...
```

**Review by boubou19 — APPROVED** [review]

**Pull request merged**